akka-typed/src/main/scala/akka/typed/javadsl/BehaviorBuilder.scala

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.typed.javadsl
+
+import scala.annotation.tailrec
+import akka.japi.pf.FI
```

Comment:
Yeah, we historically made a mess there... some of these did not have variance annotations or throws things... Not sure we can make them deprecated since they may be used in existing non deprecated APIs and we can't add overloads since that will break the target typing of java lambdas... would be good to make a ticket for this - can you open one? About unifying all the functiona types.